### PR TITLE
Add the option for changing nginx config using values.yaml and remove…

### PR DESCRIPTION
**Pull request by danielezer**
… outdated reverse proxy documentation for artifactory-ha

#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md

**What this PR does / why we need it**:
Add the option to provide nginx configuration with a values file

### CODE REVIEW

File: stable/artifactory-ha/ReverseProxyConfiguration.md [hunk 1]
```diff
@@ -10,33 +10,81 @@ the required configuration snippet which you can then download and install direc
...
+Use a values.yaml file for changing the value of nginx.mainConf or nginx.artifactoryConf
+These configuration will be mounted to the nginx container using a configmap.
+For example:
+1. Create a values file `nginx-values.yaml` with the following values:
+```yaml
+nginx:
+  mainConf: |
+    ## add HA entries when ha is configure
+    ## Replace server names with Artifactory service names (primary and member service names)
+    upstream artifactory {
```
Comment (eldada) [inline stable/artifactory-ha/ReverseProxyConfiguration.md]: Our current setup does not use the `upstream`. Nginx is aware of a single service that can include members only or primary + members in the pool.
This is controlled by key `artifactory.service.pool`. I'd like to keep it like this (unless you see a problem with it).

File: stable/artifactory-ha/ReverseProxyConfiguration.md [hunk 2]
```diff
@@ -10,33 +10,81 @@ the required configuration snippet which you can then download and install direc
...
+        
+        if ($http_x_forwarded_proto = '') {
+            set $http_x_forwarded_proto  $scheme;
+        }
+        ## Application specific logs
+        ## access_log /var/log/nginx/jfrog.team-access.log timing;
+        ## error_log /var/log/nginx/jfrog.team-error.log;
+        rewrite ^/$ /artifactory/webapp/ redirect;
+        rewrite ^/artifactory/?(/webapp)?$ /artifactory/webapp/ redirect;
+        rewrite ^/(v1|v2)/(.*) /artifactory/api/docker/$repo/$1/$2;
```
Comment (eldada) [inline stable/artifactory-ha/ReverseProxyConfiguration.md]: Add the 
```
      if ( $repo != "" ) {
        rewrite ^/(v1|v2)/(.*) /artifactory/api/docker/$repo/$1/$2;
      }
```
block

File: stable/artifactory-ha/ReverseProxyConfiguration.md [hunk 3]
```diff
@@ -10,33 +10,81 @@ the required configuration snippet which you can then download and install direc
...
+        chunked_transfer_encoding on;
+        client_max_body_size 0;
+        location /artifactory/ {
```
Comment (eldada) [inline stable/artifactory-ha/ReverseProxyConfiguration.md]: Please indent all keys in `location` block

File: stable/artifactory-ha/ReverseProxyConfiguration.md (hunk 1 above)
Comment (danielezer) [reply stable/artifactory-ha/ReverseProxyConfiguration.md]: yeah sure. I just thought I would keep it as it was before. Changed

File: stable/artifactory-ha/ReverseProxyConfiguration.md (hunk 3 above)
Comment (danielezer) [reply stable/artifactory-ha/ReverseProxyConfiguration.md]: Done

File: stable/artifactory-ha/ReverseProxyConfiguration.md (hunk 2 above)
Comment (danielezer) [reply stable/artifactory-ha/ReverseProxyConfiguration.md]: Done

File: stable/artifactory-ha/ReverseProxyConfiguration.md [hunk 4]
```diff
@@ -10,33 +10,75 @@ the required configuration snippet which you can then download and install direc
...
+    ssl_protocols TLSv1 TLSv1.1 TLSv1.2 TLSv1.3;
+    ssl_certificate  {{ .Values.nginx.persistence.mountPath }}/ssl/tls.crt;
+    ssl_certificate_key  {{ .Values.nginx.persistence.mountPath }}/ssl/tls.key;
+    ssl_session_cache shared:SSL:1m;
+    ssl_prefer_server_ciphers   on;
+    ## server configuration
+    server {
+      listen 443 ssl;
+      listen 80;
+      server_name ~(?<repo>.+)\.artifactory-artifactory artifactory-artifactory;
```
Comment (eldada) [inline stable/artifactory-ha/ReverseProxyConfiguration.md]: Change `artifactory-artifactory` to `artifactory` and add a comment above to set the actual server name used to access the LB.

Comment (danielezer) [reply stable/artifactory-ha/ReverseProxyConfiguration.md]: @eldada Done

**Comment by danielezer** [comment]
Thanks @rimusz, reverted

**Comment by btashton** [comment]
Verified the rewrite rules for docker api with the updated default nginx config.   Was able to authenticate and access images via both flows using the same configuration.

File: stable/artifactory-ha/ReverseProxyConfiguration.md [hunk 5]
```diff
@@ -10,33 +10,76 @@ the required configuration snippet which you can then download and install direc
...
+      if ($http_x_forwarded_proto = '') {
+        set $http_x_forwarded_proto  $scheme;
+      }
+      ## Application specific logs
+      ## access_log /var/log/nginx/artifactory-access.log timing;
+      ## error_log /var/log/nginx/artifactory-error.log;
+      rewrite ^/$ /artifactory/webapp/ redirect;
+      rewrite ^/artifactory/?(/webapp)?$ /artifactory/webapp/ redirect;
+      if ( $repo != "" ) {
+        rewrite ^/(v1|v2)/(.*) /artifactory/api/docker/$repo/$1/$2;
```
Comment (eldada) [inline stable/artifactory-ha/ReverseProxyConfiguration.md]: Add the lines you added in `values.yaml`

Comment (danielezer) [reply stable/artifactory-ha/ReverseProxyConfiguration.md]: @eldada Done

**Comment by danielezer** [comment]
@rimusz Can you please review? :)

**Pull request merged**